zilliqa/src/node.rs

```diff
@@ -166,17 +200,15 @@ impl Node {
         Ok(())
     }
 
-    pub fn handle_timeout(&mut self) -> Result<()> {
-        match self.consensus.timeout() {
-            Ok((leader, new_view)) => {
-                self.message_sender
-                    .send_external_message(leader, ExternalMessage::NewView(Box::new(new_view)))?;
-            }
-            Err(_) => {
-                warn!("timeout failed");
-            }
+    // handle timeout - true if something happened
```

Comment:
the timeouts are changed - the node launcher just pings consensus every time there aren't messages for a little while. It is up to consensus to decide whether to send a message (newView) given that it knows what time the last timeout was.

This moves the consensus logic more into consensus which seems correct, and tests are able to ping the timeout until something happens, when testing network failures.